lib/src/map/gestures/map_interactive_viewer.dart

```diff
@@ -0,0 +1,268 @@
+import 'package:flutter/gestures.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/src/map/gestures/services/base_services.dart';
+
+class MapInteractiveViewer extends StatefulWidget {
+  final ChildBuilder builder;
+  final MapControllerImpl controller;
+
+  const MapInteractiveViewer({
+    super.key,
+    required this.builder,
+    required this.controller,
+  });
+
+  @override
+  State<MapInteractiveViewer> createState() => MapInteractiveViewerState();
+}
+
+class MapInteractiveViewerState extends State<MapInteractiveViewer>
+    with TickerProviderStateMixin {
+  TapGestureService? _tap;
+  LongPressGestureService? _longPress;
+  SecondaryTapGestureService? _secondaryTap;
+  SecondaryLongPressGestureService? _secondaryLongPress;
+  TertiaryTapGestureService? _tertiaryTap;
+  TertiaryLongPressGestureService? _tertiaryLongPress;
+  DoubleTapGestureService? _doubleTap;
+  ScrollWheelZoomGestureService? _scrollWheelZoom;
+  TwoFingerGesturesService? _twoFingerInput;
+  DragGestureService? _drag;
+  DoubleTapDragZoomGestureService? _doubleTapDragZoom;
+  CtrlDragRotateGestureService? _ctrlDragRotate;
+
+  MapCamera get _camera => widget.controller.camera;
+
+  MapOptions get _options => widget.controller.options;
+
+  InteractionOptions get _interactionOptions => _options.interactionOptions;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.controller.interactiveViewerState = this;
+    widget.controller.addListener(reload);
+
+    // callback gestures for the application
+    if (_options.onTap != null) {
+      _tap = TapGestureService(controller: widget.controller);
+    }
+    if (_options.onLongPress != null) {
+      _longPress = LongPressGestureService(controller: widget.controller);
+    }
+    if (_options.onSecondaryTap != null) {
+      _secondaryTap = SecondaryTapGestureService(controller: widget.controller);
+    }
+    if (_options.onSecondaryLongPress != null) {
+      _secondaryLongPress =
+          SecondaryLongPressGestureService(controller: widget.controller);
+    }
+    if (_options.onTertiaryTap != null) {
+      _tertiaryTap = TertiaryTapGestureService(controller: widget.controller);
+    }
+    if (_options.onTertiaryLongPress != null) {
+      _tertiaryLongPress =
+          TertiaryLongPressGestureService(controller: widget.controller);
+    }
+    // gestures that change the map camera
+    updateGestures(null, _interactionOptions.enabledGestures);
+  }
+
+  @override
+  void dispose() {
+    widget.controller.removeListener(reload);
+    super.dispose();
+  }
+
+  void reload() {
```

Comment:
When would you override the `setState` function of `MapInteractiveViewerState`?
If you do and have a good reason to so I think the reload function would require to use the overridden function anyway.

Reply:
I'm just looking and seeing that `reload` is just calling `setState` if it's safe to. Therefore to hide that bit of abstraction from people looking at the code, and prevent them from having to check what `reload` does, it might make sense. Maybe I'm just overcomplicating things - let's leave it for now.